packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -282,6 +288,14 @@ export class WaferMap<
         return this.diesTable !== undefined;
     }
 
+    public async setData(tableData: Table): Promise<void> {
+        if (this.currentTask !== undefined) {
+            await this.currentTask;
+        }
+        this.diesTable = tableData;
+        this.waferMapUpdateTracker.queueUpdate();
```

Comment:
Conceptually we want this method to allow a user to set data and reliably wait for rendering to have finished. And it should be our primary entry point. So like the following:

1. Take either an Arrow table or old Wafer dies format
2. Assign to the right property (`diesTable` or `dies`)
3. Wait for that property update to be batched and queued ([similar to the table](https://github.com/ni/nimble/blob/d37ca77cda0729b42f05b4bac79be8ccc98ceb22/packages/nimble-components/src/table/index.ts#L280))
4. Wait for the `currentTask` created from the update to finish so that we know the side effect of setting the data has been resolved

Then we can avoid the `render-finished` event because we have a reliable way to know when the render associated with a `setData` is finished.